cmd/tar2ext4/tar2ext4.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/Microsoft/hcsshim/ext4/tar2ext4"
+)
+
+var (
+	input   = flag.String("i", "", "input file")
+	output  = flag.String("o", "", "output file")
+	overlay = flag.Bool("overlay", false, "produce overlayfs-compatible layer image")
+	vhd     = flag.Bool("vhd", false, "add a VHD footer to the end of the image")
+)
+
+func main() {
+	flag.Parse()
+	if flag.NArg() != 0 || len(*output) == 0 {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	err := func() (err error) {
+		in := os.Stdin
+		if *input != "" {
+			in, err = os.Open(*input)
+			if err != nil {
+				return err
+			}
+		}
+		out, err := os.Create(*output)
```

Comment:
Should we write to `stdout` by default? That way the caller can control the in/out stream acl's

Reply:
Well, we need to be able to seek, and stdout isn't guaranteed to be able to seek (e.g. if it's a pipe). So I was hesitant to allow this.